scientific_library/tvb/simulator/models/reduced_wong_wang_exc_io.py

```diff
@@ -0,0 +1,320 @@
+#
+# (c) 2012-2017, Baycrest Centre for Geriatric Care ("Baycrest") and others
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this
+# program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+#   CITATION:
+# When using The Virtual Brain for scientific publications, please cite it as follows:
+#
+#   Paula Sanz Leon, Stuart A. Knock, M. Marmaduke Woodman, Lia Domide,
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+
+"""
+Models based on Wong-Wang's work.
+.. moduleauthor:: Dionysios Perdikis <dionysios.perdikis@charite.de>
+
+
+"""
+
+from numba import guvectorize, float64
+from tvb.simulator.models.wong_wang import ReducedWongWang as TVBReducedWongWang
+from tvb.simulator.models.base import numpy, ModelNumbaDfun
+from tvb.basic.neotraits.api import NArray, Final, List, Range
+
+
+@guvectorize([(float64[:],)*11], '(n),(m)' + ',()'*8 + '->(n)', nopython=True)
+def _numba_update_non_state_variables(S, c, a, b, d, w, jn, r, g, io, newS):
+    "Gufunc for reduced Wong-Wang model equations."
+
+    newS[0] = S[0]  # S
+    newS[1] = S[1]  # Rint
+
+    cc = g[0]*jn[0]*c[0]
+    newS[4] = w[0] * jn[0] * S[0] + io[0] + cc  # I
+    if r[0] <= 0.0:
+        # R computed in TVB
+        x = a[0]*newS[4] - b[0]
+        h = x / (1 - numpy.exp(-d[0]*x))
+        newS[2] = h     # R
+        newS[3] = 0.0   # Rin
+    else:
+        # R updated from Spiking Network model
+        # Rate has to be scaled down from the Spiking neural model rate range to the TVB one
+        if S[1] < 40.0:  # Rint
+            # For low activity
+            R = numpy.sqrt(S[1])
+        else:
+            # For high activity
+            R = 0.0000050 * g[0] * S[1] ** 2
+        newS[2] = R
+        newS[3] = S[3]  # Rin
+
+
+@guvectorize([(float64[:],)*6], '(n)' + ',()'*4 + '->(n)', nopython=True)
+def _numba_dfun(S, g, t, r, tr, dx):
+    "Gufunc for reduced Wong-Wang model equations."
+    if r[0] > 0.0:
+        # Integrate rate from Spiking Network
+        # Rint
+        dx[1] = (- S[1] + S[3]) / tr[0]
+    else:
+        # TVB computation
+        dx[1] = 0.0
+    dx[0] = - (S[0] / t[0]) + (1.0 - S[0]) * S[2] * g[0]   # S
+    dx[2] = 0.0  # R
+    dx[3] = 0.0  # Rin
+    dx[4] = 0.0  # I
+
+
+class ReducedWongWangExcIO(TVBReducedWongWang):
```

Comment:
This new model, if remaining here should be included in the normal tvb-library unit-tests round (where all the other unit-tests are covered)